hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMSecurityProtocolServer.java

```diff
@@ -403,13 +415,20 @@ public List<String> listCACertificate() throws IOException {
   }
 
   @Override
-  public String getRootCACertificate() throws IOException {
-    LOGGER.debug("Getting Root CA certificate.");
-    if (storageContainerManager.getScmStorageConfig()
-        .checkPrimarySCMIdInitialized()) {
-      return CertificateCodec.getPEMEncodedString(rootCACertificate);
+  public synchronized String getRootCACertificate() throws IOException {
+    Date lastCertDate = new Date(0);
+    X509Certificate lastCert = null;
+    for (X509Certificate cert : rootCACertificate) {
+      if (cert.getNotAfter().after(lastCertDate)) {
+        lastCertDate = cert.getNotAfter();
+        lastCert = cert;
+      }
     }
-    return null;
+    return CertificateCodec.getPEMEncodedString(lastCert);
+  }
+
+  public synchronized void addNewRootCa(X509Certificate rootCaCertToAdd) {
```

Comment:
I can see this is used only by test code, can you mark this with test annotation. Also synchronized is not required.

Reply:
This method is going to be used later once the root ca certificate rotation is implemented. It's here because it makes sense as well as we can ensure that the sync is properly placed.